daml/Marketplace/Trading/Order.daml

```diff
@@ -0,0 +1,83 @@
+module Marketplace.Trading.Order where
+
+import DA.Finance.Asset (AssetDeposit)
+import DA.Finance.Types (Id(..), Asset(..))
+
+data Side
+    = Buy
+    | Sell
+  deriving (Eq, Show)
+
+data OrderType
+    = Market
+    | Limit with
+        price : Decimal
+  deriving (Eq, Show)
+
+data TimeInForce
+    = GTC
+      -- ^ Good Till Cancelled (Rests on book until cancellation)
+    | GTD with
+        expiryDate : Int
+      -- ^ Good Till Date (At expiryDate, order will be automatically cancelled).
+      -- UTC date and time in seconds
+    | GAA
+      -- ^ Good At Auction (Expires after auction if not filled)
+    | IOC
+      -- ^ Immediate Or Cancel (Allows for partial fills)
+    | FOK
+      -- ^ Fill Or Kill (All or nothing)
+  deriving (Eq, Show)
+
+data OrderDetails = OrderDetails with
```

Comment:
`Details`

Reply:
👍  (I copied this from Trading.daml without updating the name - I'll change it in the next PR)